parts/linux/cloud-init/artifacts/cse_cmd.sh

```diff
@@ -1,9 +1,9 @@
 PROVISION_OUTPUT="/var/log/azure/cluster-provision-cse-output.log";
 echo $(date),$(hostname) > ${PROVISION_OUTPUT};
 {{if ShouldEnableCustomData}}
-cloud-init status --wait > /dev/null 2>&1;
-[ "$?" -ne 0 ] && echo 'cloud-init failed' >> ${PROVISION_OUTPUT} && exit 1;
-echo "cloud-init succeeded" >> ${PROVISION_OUTPUT};
+/bin/bash -c "source /opt/azure/containers/provision_source.sh; handleCloudInitStatus \"${PROVISION_OUTPUT}\"; returnStatus=\$?; echo \"Cloud init status check exit code: \$returnStatus\" >> ${PROVISION_OUTPUT}; exit \$returnStatus" >> ${PROVISION_OUTPUT} 2>&1;
```

Comment:
be careful about sourcing entire provisioning scripts like this, anything not defined in a function will be executed via the source cmd